crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class RepositoriesController < ApplicationController
+  before_filter :load_registry
```

Comment:
> It's because adam decided against some kind of hook or filter system within the fake models. And it's because adam suggested this kind of design...

1. I didn't "decide" anything, because I am not the maintainer / overlord / dictator of Crowbar (or any other part of the product).  I suggest things and try to help reach a consensus within the team.
2. You are completely misrepresenting my opinion.  [I originally stated I was against this `before_filter`](https://github.com/crowbar/crowbar-core/pull/18#discussion_r38846328), I did not suggest it, and also I did *not* say I was against some kind of hook or filter system within the model, I simply said that ["you can't have a `before_filter` in `Crowbar::Repository`; that's a Rails controller thing"](https://github.com/crowbar/crowbar-core/pull/18#discussion_r38846397).  However that was all before [this conversation](https://github.com/crowbar/crowbar-core/pull/18#issuecomment-138387716) which I think probably contains the crucial info @vuntz may be missing regarding how class variable caching works, i.e. it persists (IIUC) per thread across requests, which breaks normal memoization, or JIT-based loading.  So maybe loading once per request via `before_filter` is the safest and most efficient approach after all, even though I still don't like that it violates the Law of Demeter.

I'm really surprised this PR was merged with unfinished parts, especially without two +1s.